.github/workflows/test.yml

```diff
@@ -112,8 +112,17 @@ jobs:
           name: k-framework
           authToken: '${{ secrets.CACHIX_PUBLIC_TOKEN }}'
 
-      - name: Build
-        run: GC_DONT_GC=1 nix build .#kore-exec .#kore-rpc-booster
+      - name: 'Build and Cache'
+        uses: workflow/nix-shell-action@v3
+        env:
+          GC_DONT_GC: 1
+          CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_PUBLIC_TOKEN }}
+        with:
+          packages: jq
+          script: |
+            HASKELL_BACKEND=$(nix build --extra-experimental-features 'nix-command flakes' .#kore-exec .#kore-rpc-booster --json | jq -r '.[].outputs | to_entries[].value')
+            DRV=$(nix-store --query --deriver ${HASKELL_BACKEND})
+            nix-store --query --requisites --include-outputs ${DRV} | cachix push k-framework
```

Comment:
(as Everett also asked on slack)
Not sure we should push to cachix from PR builds. We have the cachix action (currently only on linux but could probably add it as a noop to the Mac build without issues) so the build artefacts _should_ already be pushed up to accelerate subsequent builds without this change.